File: src/frontend/src/app/components/financeiro/condicao-pagamento/model/condicao-pagamento.dto.ts
import { Exclude, Expose, plainToInstance } from 'class-transformer';

@Exclude()
export class CondicaoPagamentoDTO {
  @Expose()
  id?: string;

  @Expose()
  condicao: string;

  @Expose()
  descricao?: string;

  @Expose()
  ativo: boolean;

  @Expose()
  quantidadeParcelas?: number;

  @Expose()
  diasVencimento?: number[];

  constructor(
    condicao: string,
    ativo: boolean,
    descricao?: string,
    quantidadeParcelas?: number,
    diasVencimento?: number[],
    id?: string
  ) {
    this.condicao = condicao;
    this.ativo = ativo;
    this.descricao = descricao;
    this.quantidadeParcelas = quantidadeParcelas;
    this.diasVencimento = diasVencimento;
    this.id = id;
  }

  static from(data: Partial<CondicaoPagamentoDTO>): CondicaoPagamentoDTO {
    return plainToInstance(CondicaoPagamentoDTO, data);
  }
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/condicaopagamento/CondicaoPagamentoDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.condicaopagamento;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class CondicaoPagamentoDTO implements DTO {

    private UUID id;
    private String condicao;
    private String descricao;
    private Boolean ativo;
    private Integer quantidadeParcelas;
    private List<Integer> diasVencimento;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
